lld/MachO/Driver.cpp

```diff
@@ -1838,34 +1838,44 @@ bool link(ArrayRef<const char *> argsArr, llvm::raw_ostream &stdoutOS,
       if (const Arg *arg = args.getLastArgNoClaim(OPT_call_graph_profile_sort))
         error(firstArgStr + " is incompatible with " + arg->getSpelling());
   };
-  if (const Arg *arg = args.getLastArg(OPT_irpgo_profile_sort)) {
-    config->irpgoProfileSortProfilePath = arg->getValue();
+  if (args.hasArg(OPT_irpgo_profile_sort) ||
+      args.hasArg(OPT_irpgo_profile_sort_eq))
+    error("--irpgo-profile-sort is deprecated. Please use "
+          "--bp-startup-sort=function");
+  if (const Arg *arg = args.getLastArg(OPT_irpgo_profile_eq)) {
+    config->irpgoProfilePath = arg->getValue();
     IncompatWithCGSort(arg->getSpelling());
```

Comment:
I'm open to discussion, but I think `-irpgo-profile=` could be compatible with CGSort. It's only if we also have `-bp-start-sort=function` where this is a problem, which is checked later.
```suggestion
```